File: python/valid_palindrome.py
def is_palindrome(word:str):
    l=0
    r=len(word)-1

    while l<r:
        if word[l]==word[r]:
            l=l+1
            r=r-1
        else:
            return False
        
    return True

if __name__ == "__main__":
    print(is_palindrome("noon"))
    print(is_palindrome("civic"))
    print(is_palindrome("madam"))
    print(is_palindrome("racecar"))
    print(is_palindrome("soon"))

File: java/ValidPalindrome.java
public class ValidPalindrome {
    public static boolean isPalindrome(String word) {
        int left = 0;
        int right = word.length() - 1;
        while(left < right) {
            if(word.charAt(left) == word.charAt(right)) {
                left++;
                right--;
            }
            else {
                return false;
            }
        }
        return true;
    }
    public static void main(String[]args) {
        System.out.println(isPalindrome("racecar"));
		System.out.println(isPalindrome("madam"));
        System.out.println(isPalindrome("noon"));
        System.out.println(isPalindrome("soon"));
    }
}
